components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_iat_claims_loader.c

```diff
@@ -0,0 +1,225 @@
+    psa_hash_operation_t hash_handle;
+
+    crypto_ret = psa_open_key(PSA_KEY_LIFETIME_PERSISTENT, key_id, &handle);
+    if (crypto_ret != PSA_SUCCESS) {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_get_key_information(handle, &type, &bits);
+    if (crypto_ret != PSA_SUCCESS) {
+        status = TFM_PLAT_ERR_SYSTEM_ERR;
+        goto exit;
+    }
+    if (!PSA_KEY_TYPE_IS_ECC(type)) {
+        status = TFM_PLAT_ERR_SYSTEM_ERR;
+        goto exit;
+    }
+    public_type = PSA_KEY_TYPE_PUBLIC_KEY_OF_KEYPAIR(type);
+    public_key_size = PSA_KEY_EXPORT_MAX_SIZE(public_type, bits);
+    public_key = (uint8_t *) malloc(public_key_size);
+    if (public_key == NULL) {
+        status = TFM_PLAT_ERR_SYSTEM_ERR;
+        goto exit;
+    }
+
+    crypto_ret = psa_export_public_key(handle,
+                                       public_key, public_key_size,
+                                       &public_key_length);
+    if (crypto_ret != PSA_SUCCESS) {
+        status = TFM_PLAT_ERR_SYSTEM_ERR;
+        goto exit;
+    }
+
+    crypto_ret = psa_hash_setup(&hash_handle, PSA_ALG_SHA_256);
+    if (crypto_ret != PSA_SUCCESS) {
+        status = TFM_PLAT_ERR_SYSTEM_ERR;
+        goto exit;
+    }
+
+    psa_hash_update(&hash_handle, public_key, public_key_length);
+
+    crypto_ret = psa_hash_finish(&hash_handle,
+                                 buf,
+                                 *size,
+                                 (size_t *) size);
+    if (crypto_ret != PSA_SUCCESS) {
+        status = TFM_PLAT_ERR_SYSTEM_ERR;
+        goto exit;
+    }
+
+exit:
+    if (public_key != NULL) {
+        free(public_key);
+    }
+    psa_close_key(handle);
+    return status;
+}
+
+/**
+ * \brief Copy the device specific ID to the destination buffer
+ *
+ * \param[out]  p_dst  Pointer to buffer where to store ID
+ * \param[in]   p_src  Pointer to the ID
+ * \param[in]   size   Length of the ID
+ */
+static inline void copy_id(uint8_t *p_dst, uint8_t *p_src, size_t size)
+{
+    uint32_t i;
+
+    for (i = size; i > 0; i--) {
+        *p_dst = *p_src;
+        p_src++;
+        p_dst++;
+    }
+}
+
+enum psa_attest_err_t attest_get_caller_client_id(int32_t *caller_id)
+{
+    *caller_id = g_caller_id;
+    return PSA_ATTEST_ERR_SUCCESS;
+}
+
+/* Boot seed data is part of bootloader status*/
+enum tfm_plat_err_t tfm_plat_get_boot_seed(uint32_t size, uint8_t *buf)
+{
+    return PSA_ATTEST_ERR_CLAIM_UNAVAILABLE;
+}
+
+/**
+ * Instance ID is mapped to EAT Universal Entity ID (UEID)
+ * This implementation creates the instance ID as follows:
+ *  - byte 0:    0x01 indicates the type of UEID to be GUID
+ *  - byte 1-32: Hash of attestation public key. Public key is hashed in raw
+ *               format without any encoding.
+ */
+enum tfm_plat_err_t tfm_plat_get_instance_id(uint32_t *size, uint8_t *buf)
+{
+    enum tfm_plat_err_t status;
+    uint8_t *p_dst;
+    uint8_t p_src[ATTEST_PUB_KEY_SHA_256_SIZE];
```

Comment:
There's no need for this extra buffer - nothing's stopping you outputting straight into the final destination. (The thing you're calling could length limit if necessary, but you don't need it to anyway).

Reply:
Right, Fixed